python/cuml/svm/svm.pyx

```diff
@@ -0,0 +1,297 @@
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+import ctypes
+import cudf
+import numpy as np
+
+from numba import cuda
+
+from cython.operator cimport dereference as deref
+from libc.stdint cimport uintptr_t
+
+from cuml.common.base import Base
+from cuml.common.handle cimport cumlHandle
+from cuml.utils import input_to_dev_array, zeros, get_cudf_column_ptr
+
+from sklearn.exceptions import NotFittedError
+
+cdef extern from "gram/kernelparams.h" namespace "MLCommon::GramMatrix":
+    enum KernelType:
+        LINEAR, POLYNOMIAL, RBF, TANH
+
+    cdef cppclass KernelParams:
+        KernelType kernel
+        int degree
+        double gamma
+        double coef0
+        KernelParams(KernelType kernel, int degree, double gamma, double coef0)
+
+cdef extern from "svm/svc.h" namespace "ML::SVM":
+
+    cdef cppclass CppSVC "ML::SVM::SVC" [math_t]:
+        # The CppSVC class manages the memory of the parameters that are found
+        # during fitting (the support vectors, and the dual_coefficients). The
+        # number of these parameters are not known before fitting.
+        int n_support
+        math_t *dual_coefs
+        int *support_idx
+        math_t b
+        KernelParams kernel_params
+        math_t C
+        math_t tol
+
+        CppSVC(cumlHandle& handle, math_t C, math_t tol,
+               KernelParams kernel_params, float cache_size,
+               int max_iter) except+
+        void fit(math_t *input, int n_rows, int n_cols, math_t *labels) except+
+        void predict(math_t *input, int n_rows, int n_cols,
+                     math_t *preds) except+
+
+
+class SVC(Base):
+    """
+    SVC (C-Support Vector Classification)
+
+    Currently only binary classification is supported.
+
+    Parameters
+    ----------
+    handle : cuml.Handle
+        If it is None, a new one is created for this class
+    C : float (default = 1.0)
+        Penalty parameter C
+    kernel : string (default='linear' TODO change to 'rbf' once implemented)
+        Specifies the kernel function. Possible options: 'linear', 'poly',
+        'rbf', 'sigmoid', 'precomputed'
+    degree : int (default=3)
+        Degree of polynomial kernel function.
+    gamma : float (default = 'auto')
+        Coefficient for rbf, poly, and sigmoid kernels.
+    coef0 : float (default = 0.0)
+        Independent term in kernel function, only signifficant for poly and
+        sigmoid
+    tol : float (default = 1e-3)
+        Tolerance for stopping criterion.
+    cache_size : float (default = 200 MiB)
+        Size of the kernel cache n MiB (TODO)
+    max_iter : int (default = 100*n_samples)
+        Limit the number of outer iterations in the solver
+    verbose : bool (default = False)
+        verbose mode
+
+    Attributes
+    ----------
+
+    n_support_ : int
+        The total number of support vectors.
+        TODO change this to represent number support vectors for each class.
```

Comment:
I might be mistaken, but I believe this TODO will become part of the documentation. If this is the case, can we move this somewhere else? 

Add github issue and link it, where it moves.